src/SonarScanner.MSBuild.PreProcessor/WebService/SonarWebService.cs

```diff
@@ -349,43 +244,64 @@ private async Task<T> ExecuteWithLogs<T>(Func<Task<T>> request, Uri logUri)
             return settings;
         }
 
-        private void GetPropertyValue(Dictionary<string, string> settings, JToken p)
+        /// <summary>
+        /// Concatenates project key and branch into one string.
+        /// </summary>
+        /// <param name="projectKey">Unique project key</param>
+        /// <param name="projectBranch">Specified branch of the project. Null if no branch to be specified.</param>
+        /// <returns>A correctly formatted branch-specific identifier (if appropriate) for a given project.</returns>
+        protected Uri GetUri(string query, params string[] args) =>
+            new(serverUri, Escape(query, args));
+
+        protected virtual Uri AddOrganization(Uri uri, string organization) =>
```

Comment:
Further improvement for another PR: This method should not have `organization` parameter. This whole class should remember organization from constructor and add it without forcing everyone to pass it explicitly.

The organization is known at the time of creation, global and immutable at the run of the whole application.

Reply:
Not in the scope of this PR. We can handle this later.

Reply:
#1476